src/main/java/com/faforever/api/leaderboard/LeaderboardService.java

```diff
@@ -30,11 +30,21 @@ public LeaderboardService(GlobalLeaderboardRepository globalLeaderboardRepositor
     return ladder1v1LeaderboardRepository.getLeaderboardByPage(getPageable(page, pageSize));
   }
 
+  @Cacheable(LEADERBOARD_RANKED_1V1_CACHE_NAME)
+  public Page<Ladder1v1LeaderboardEntry> getLadder1v1LeaderboardAndFilterPlayerByRegex(@Nullable Integer page, @Nullable Integer pageSize, @NotNull String playNameRgex) {
+    return ladder1v1LeaderboardRepository.getLeaderboardByPageAndPlayerNameMatchesRegex(getPageable(page, pageSize), playNameRgex);
+  }
+
   @Cacheable(LEADERBOARD_GLOBAL_CACHE_NAME)
   public Page<GlobalLeaderboardEntry> getGlobalLeaderboard(@Nullable Integer page, @Nullable Integer pageSize) {
     return globalLeaderboardRepository.getLeaderboardByPage(getPageable(page, pageSize));
   }
 
+  @Cacheable(LEADERBOARD_GLOBAL_CACHE_NAME)
```

Comment:
I don't think caching makes sense here. Also if it does I don't think you should use the same for the cache key here.